backend/src/main.ts

```diff
@@ -203,14 +216,34 @@ export const moodLog = async (req: UserRequest, res: Response) => {
     let globalNow = DateTime.utc();
 
     if (data.addFlag) {
-        if (typeof data.addFlag !== "string" || !data.addFlag.startsWith("summary:")) {
+        if (typeof data.addFlag !== "string") {
+            console.log("Invalid addFlag:", data.addFlag);
             res.send(400);
             return;
         }
 
-        globalNow = DateTime.fromISO(data.addFlag.split(":")[1], { zone: data.timezone });
+        if (data.addFlag.startsWith("summary:")){
+            const timestamp = data.addFlag.split("summary:")[1].split(" ")[0];
+            if (!timestamp || isNaN(Date.parse(timestamp))) {
+                console.log("Invalid summary timestamp:", timestamp);
+                res.send(400);
+                return;
+            }
+            globalNow = DateTime.fromISO(data.addFlag.split("summary:")[1], {
+                zone: data.timezone,
+            });
+        } else if (data.addFlag.includes("offlineSync:")) {
+            let timestamp = data.addFlag.split("offlineSync:")[1];
+            if (!timestamp || isNaN(Number(timestamp)) || Number(timestamp) < 0) {
```

Comment:
Can we bound this more? Like instead of < 0 maybe like in the last year or something lol

Reply:
fixed